backend/api/Controllers/InspectionFindingsController.cs

```diff
@@ -0,0 +1,95 @@
+            _inspectionService = inspectionService;
+
+        }
+
+        /// <summary>
+        /// Associate a new inspection finding with the inspection corresponding to isarStepId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpPost("add-findings")]
+        [Authorize(Roles = Role.Admin)]
+        [ProducesResponseType(typeof(InspectionFindings), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<InspectionFindings>> AddFindings([FromBody] InspectionFindingsQuery inspectionFinding)
+        {
+            _logger.LogInformation("Updating inspection findings for inspection with isarStepId '{Id}'", inspectionFinding.IsarStepId);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(inspectionFinding.IsarStepId);
+                if (inspection != null)
+                {
+                    inspection = await _inspectionService.AddFindings(inspectionFinding);
+                    return Ok(inspection);
+                }
+
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error while adding findings to inspection with IsarStepId '{inspectionFinding.IsarStepId}'", inspectionFinding.IsarStepId);
+                throw;
+            }
+            return NotFound();
+        }
+
+        /// <summary>
+        /// Get the full inspection against an isarStepId
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        [HttpGet]
+        [Authorize(Roles = Role.Admin)]
+        [Route("{id}")]
+        [ProducesResponseType(typeof(AreaResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status409Conflict)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<InspectionFindings>> GetInspections([FromRoute] string id)
+        {
+            _logger.LogInformation("Get inspection by ID '{inspectionFinding.InspectionId}'", id);
+            try
+            {
+                var inspection = await _inspectionService.ReadByIsarStepId(id);
+                if (inspection != null)
+                {
+                    return Ok(inspection);
+                }
+
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Error while finding an inspection with inspection id '{id}'", id);
+                throw;
```

Comment:
Should return InternalServerError or similar

Reply:
Done now.